### DISCUSSION TITLE
Making minimap view background semi transparent in multi-view example from deck.gl

### DISCUSSION THREAD

**Discussion by argilya** [visgl/deck.gl#7580] [Q&A] [answered]
Hello,

I have a question about the [multi-view Radio example](https://deck.gl/examples/multi-view/) on deck.gl's website.

The application renders two views:
1) A main view
2) A minimap view

In the example, the minimap view is set to `clear: true`, which clears the main view layers from under the minimap. The minimap view also encloses an html div with a white background that hides the base map. So it looks like this:

![image](https://user-images.githubusercontent.com/100630262/212740956-195e50b4-a66d-441e-9103-aa8fda94f77a.png)

I would like make the minimap background semi-transparent, so you can see both the map and the main view behind the minimap.

Is that possible with the current set up? My gut feeling is no. Both views are rendered onto the same canvas, so you can't insert the background div between the two. Here is what it looks like when you set the minimap view to `clear: false`, and color of the background div to black at 50% alpha. You can see the main view layers being drawn on top of the background div.

![image](https://user-images.githubusercontent.com/100630262/212741781-b95cf435-71ab-4d5f-a6c8-51e77d4e99ac.png)

I'd like the main view to be drawn behind the transparent minimap background. One solution I can think of is mount the minimap view in its own DeckGL component, so the two layers are drawn on separate canvases, but I'd rather not do that.

Any thoughts? Thank you!

**Comment by ibgreen** [comment]
Quick though - In some other places (at least in luma.gl) `clear` can be set to a color `[r,g,b,a]` instead of a boolean. Not sure if it is supported here, but if that works, choose an appropriate alpha value and drop the backing HTML div?

**Reply by argilya** [reply]
Hello,

Thank you for the suggestion. You are right about the `clear` API. The [docs](https://deck.gl/docs/api-reference/core/view#clear) say it can accept an object with key `color` that will be used for clearing.

However, applying a non-false `clear` erases the main view layers. Here is what it looks like with `clear: { color: [0, 0, 0, 0.5] }` and with no HTML background div:

![image](https://user-images.githubusercontent.com/100630262/212951906-2d0a3a54-5faf-4926-9ebc-bef480e1d054.png)

There are two other optional keys on the `clear` object, `depth` and `stencil`, which are booleans. I played around with them, and the result is the same regardless of what they are set to.

**Reply by ibgreen** [reply]
Yes that makes sense. If all else fails, I suppose you could always create a layer with a polygon that covers the screen in the desired color, and filter that out in the main viewport.

**Reply by argilya** [reply]
That's pretty clever, and it worked like a charm. Thanks!

![image](https://user-images.githubusercontent.com/100630262/212970263-28530810-4dd4-4419-bc9c-504063cd066e.png)

**Comment by argilya** [comment, marked as answer]
Resolved by creating an additional layer as suggested in https://github.com/visgl/deck.gl/discussions/7580#discussioncomment-4708557 by @ibgreen.
